t/op/magic.t

```diff
@@ -433,6 +433,23 @@ EOP
   }
 }
 
+# Check that assigning to $0 properly handles upgraded strings:
+{
+  my $eacute = "\N{U+00e9}";
+  utf8::encode($eacute);
+  utf8::upgrade($eacute);
+
+  $0 = $eacute;
+  is ($0, $eacute, 'checked $0');
+
+  SKIP: {
+    skip "Test is for Linux, not $^O" if $^O ne 'linux';
+
+    my $slurp = `cat /proc/$$/cmdline`;
+    like( $slurp, qr<\Aé\0>, '/proc cmdline shows as expected' );
```

Comment:
This also should check for the UTF-8 encoding of `\N{U+00E9}` rather than depending on the literal character.

Reply:
done